SO-4892: field selection support

cmark opened the pull request:
This PR adds a new `field` query parameter to all list and get endpoint to restrict the payload size of the returned result and make fetching of documents as fast as possible.

cmark committed:
feat(api): add support for `field` query parameter...

...on all list and get endpoints

cmark committed:
feat(api): validate selected field names using the underlying...

...document mapping

Fix couple of partial model loading issues.

cmark committed:
feat(api): support loading additional doc fields when not explicitly...

...requested but they are required to complete the search request

apeteri commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/commit/CommitInfo.java:
```diff
@@ -31,6 +32,8 @@
 
 	private static final long serialVersionUID = 1L;
 	
+	public static final List<String> DEAFULT_FIELD_SELECTION = List.of(Commit.Fields.ID, Commit.Fields.AUTHOR, Commit.Fields.BRANCH, Commit.Fields.COMMENT, Commit.Fields.TIMESTAMP, Commit.Fields.GROUP_ID);
```
These kind of "default set of fields" constants should live in the corresponding, eg. `Commit.Fields` class instead. We already have `ALL`s there which are similar.

cmark committed:
feat(api): support field selection in generic concept search api

Fix incorrect order of method calls when configuring additional fields
to load.

cmark committed:
feat(api): move CommitInfo.DEFAULT_FIELD_SELECTION constant to inner cls
